internals/cli/cmd_push.go

```diff
@@ -74,29 +74,27 @@ func (cmd *cmdPush) Execute(args []string) error {
 	}
 	defer f.Close()
 
-	st, err := f.Stat()
-	if err != nil {
+	var permissions os.FileMode
+	if cmd.Permissions != "" {
+		p, err := strconv.ParseUint(cmd.Permissions, 8, 32)
+		if err != nil {
+			return fmt.Errorf("invalid mode for directory: %q", cmd.Permissions)
+		}
+		permissions = os.FileMode(p)
+	} else if st, err := f.Stat(); err == nil {
```

Comment:
Sorry to be nit-picky, but I'd prefer we write this as follows to make the if/else variants balanced and each handling one case, rather than the stat error handling put into a 3rd clause, and to use the usual `!= nil` expression rather than `== nil`.

```go
} else {
    st, err := f.Stat()
    if err != nil {
        return err
    }
    permissions = st.Mode().Perm()
}
```

Reply:
Thanks! Will take into account from now on (: